cmd/gateway-main.go

```diff
@@ -188,17 +219,11 @@ func gatewayMain(ctx *cli.Context) {
 	backendType := ctx.Args().Get(0)
 	// Second argument is the endpoint address (optional)
 	endpointAddr := ctx.Args().Get(1)
-	// Third argument is the address flag
+
 	serverAddr := ctx.String("address")
 
-	if endpointAddr != "" {
-		// Reject the endpoint if it points to the gateway handler itself.
-		sameTarget, err := sameLocalAddrs(endpointAddr, serverAddr)
-		fatalIf(err, "Unable to compare server and endpoint addresses.")
-		if sameTarget {
-			fatalIf(errors.New("endpoint points to the local gateway"), "Endpoint url is not allowed")
-		}
-	}
+	err := validateGatewayArguments(serverAddr, endpointAddr)
+	fatalIf(err, "Unable to validate")
```

Comment:
"Unable to validate" is not correct. We did validate and found the arguments invalid. We could say something like "Invalid arguments: custom endpoint and gateway addresses are identical."